### PR TITLE
Adding commands for assumed role account switching

### PR DESCRIPTION
**Pull request by EricZaporzan**
https://docs.aws.amazon.com/IAM/latest/UserGuide/tutorial_cross-account-with-roles.html

Enabling the ability to jump to "sub" profile from an initial parent profile.

### CODE REVIEW

File: src/exporter.ts [hunk 1]
```diff
@@ -4,31 +4,8 @@ import { AWSX_HOME } from './config';
...
-    exportScript += ` AWS_SESSION_TOKEN=${sessionToken}`;
-  } else {
-    exportScript += ` AWS_SESSION_TOKEN=""`;
-  }
-
-  if (outputFormat) {
-    exportScript += ` AWS_DEFAULT_OUTPUT=${outputFormat}`;
-  }
-
-  fs.writeFileSync(EXPORTS_PATH, exportScript);
```
Comment (EricZaporzan) [inline src/exporter.ts]: So I tested it out, and the `AWS_PROFILE` option is really all that's needed; the rest of the config is drawn from the ini files. 

The AWS_SDK_LOAD_CONFIG flag is necessary for certain libraries to actually look for config in the right place.

File: src/mfa-login.ts [hunk 2]
```diff
@@ -14,6 +14,14 @@ export interface ProfileConfiguration {
   mfaSessionValid?: boolean;
 }
 
+export interface AssumeRoleProfileConfiguration {
+  profileName: string;
+  parentProfileName: string;
+  roleArn: string;
```
Comment (iansu) [inline src/mfa-login.ts]: These last three should be prefixed with `aws` to be consistent with the profile configuration.

File: src/config.ts [hunk 3]
```diff
@@ -192,6 +192,54 @@ const deleteProfile = (profileName: string): void => {
   writeConfig(AWS_CREDENTIALS_PATH, awsCredentials);
 };
 
+const getAssumeRoleProfiles = (parentProfile?: string): AssumeRoleProfileConfiguration[] => {
+  const awsConfig = getConfig(AWS_CONFIG_PATH);
+
+  const profiles: AssumeRoleProfileConfiguration[] = [];
+
+  for (const profile in awsConfig) {
```
Comment (iansu) [inline src/config.ts]: I think this should be a `for...of` loop, not a `for...in` loop.

File: src/app.ts [hunk 4]
```diff
@@ -262,6 +309,109 @@ const removeProfile = async (name?: string): Promise<void> => {
...
+    createAssumeRoleProfile(assumeRoleProfile);
+  } else {
+    const profileAnswers = await inquirer.prompt([
+      {
+        type: 'input',
+        name: 'profile',
+        message: 'Name'
+      },
+      {
+        type: 'input',
```
Comment (iansu) [inline src/app.ts]: This could be a list input so they could just select one of their existing profiles.

File: src/app.ts [hunk 5]
```diff
@@ -32,7 +40,48 @@ const validateMfaExpiry = (mfaExpiry: number): boolean | string => {
   }
 };
 
-const switchProfile = async (name?: string, forceMFA?: boolean): Promise<void> => {
+const switchToAssumeRoleProfile = async (
```
Comment (iansu) [inline src/app.ts]: I'd move this function below `switchProfile`. Also maybe the name should be `switchAssumeRoleProfile` for consistency?

Comment (EricZaporzan) [reply src/app.ts]: `switchProfile` actually depends on `switchToAssumeRoleProfile`, so the linter would throw a fit. I could change the name though.

File: src/app.ts (hunk 4 above)
Comment (EricZaporzan) [reply src/app.ts]: Good call

File: src/config.ts (hunk 3 above)
Comment (EricZaporzan) [reply src/config.ts]: Since this is an object with a key we need to access, I don't _think_ that works

File: src/app.ts [hunk 6]
```diff
@@ -32,7 +40,48 @@ const validateMfaExpiry = (mfaExpiry: number): boolean | string => {
...
+  if (assumeRoleProfileName) {
+    if (assumeRoleProfiles.includes(assumeRoleProfileName)) {
+      exportEnvironmentVariables(assumeRoleProfileName);
+
+      return assumeRoleProfileName;
+    } else {
+      console.error(chalk.red(`No profile '${assumeRoleProfileName}' found.`));
+    }
+  } else if (assumeRoleProfiles.length > 0) {
+    const rootProfileOption = 'Remain on root profile';
```
Comment (iansu) [inline src/app.ts:60]: I'd change this to just `root profile`

File: src/config.ts (hunk 3 above)
Comment (iansu) [reply src/config.ts:200]: In that case you would probably be better off using `for (const profile of Object.keys(awsConfig))`

**Pull request merged**